keps/sig-cloud-provider/20190422-cloud-controller-manager-migration.md

```diff
@@ -152,53 +157,125 @@ will prevent any of the v1.18 CCMs from claiming the lock. When the current hold
 
 ### Implementation Details/Notes/Constraints [optional]
 
-TODO after KEP summary and proposal is approved
+#### Migration Configuration
 
-### Risks and Mitigations
+The migration lock will be configured by defining new API types that will then be passed into the KCM and CCM.
 
-TODO after KEP summary and proposal is approved
+```go
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// LeaderMigrationConfiguration provides versioned configuration for all migrating leader locks.
+type LeaderMigrationConfiguration struct {
```

Comment:
I don't feel strongly about how this API is designed, I'll defer to @cheftako on this